runtime/src/main/java/org/corfudb/runtime/view/stream/AddressMapStreamView.java

```diff
@@ -83,6 +83,9 @@ protected ILogData removeFromQueue(NavigableSet<Long> queue) {
                 if (ld.containsStream(this.id) && ld.getType() == DataType.DATA) {
                     addToResolvedQueue(getCurrentContext(), currentRead, ld);
                     readNext = false;
+                } else if (ld.isCompacted()) {
```

Comment:
It's better to add ld.containsStream check here.

Reply:
The additional check is added.